File: DSA/Ayush_Mishra_590014844/Day_37_Question_2.c
#include <stdio.h>
#include <ctype.h>
#define MAX 100

int main() {
    char expr[MAX];
    int stack[MAX], top = -1;

    scanf("%s", expr);

    for(int i = 0; expr[i]; i++) {
        if(isdigit(expr[i])) {
            stack[++top] = expr[i] - '0';
        } else {
            int b = stack[top--];
            int a = stack[top--];
            if(expr[i] == '+') stack[++top] = a + b;
            else if(expr[i] == '-') stack[++top] = a - b;
            else if(expr[i] == '*') stack[++top] = a * b;
            else if(expr[i] == '/') stack[++top] = a / b;
        }
    }

    printf("%d\n", stack[top]);
    return 0;
}

File: DSA/Ayush_Mishra_590014844/Day_37_Question_2.py
def evaluate_postfix(expr):
    stack = []
    for ch in expr:
        if ch.isdigit():
            stack.append(int(ch))
        else:
            b = stack.pop()
            a = stack.pop()
            if ch == '+': stack.append(a + b)
            elif ch == '-': stack.append(a - b)
            elif ch == '*': stack.append(a * b)
            elif ch == '/': stack.append(a // b)
    return stack[-1]

expr = input().strip()
print(evaluate_postfix(expr))
